pkg/azurefile/azurefile.go

```diff
@@ -180,3 +221,36 @@ func getStorageAccount(secrets map[string]string) (string, string, error) {
 
 	return accountName, accountKey, nil
 }
+
+// File share names can contain only lowercase letters, numbers, and hyphens,
+// and must begin and end with a letter or a number,
+// and must be from 3 through 63 characters long.
+// The name cannot contain two consecutive hyphens.
+//
+// See https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-shares--directories--files--and-metadata#share-names
+func getValidFileShareName(volumeName string) string {
+	fileShareName := strings.Replace(volumeName, "--", "-", -1)
```

Comment:
remove this and move following code into the end of this function:
```
fileShareName = strings.Replace(fileShareName, "--", "-", -1)
```
so we only need one Replace action.

Reply:
If in this way, in the warning logging, user maybe get a wrong filesShare name.

Reply:
> If in this way, in the warning logging, user maybe get a wrong filesShare name.

I think that's ok, it's only a log

Reply:
done